pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -594,33 +588,28 @@ func (c *ClusterMemberController) getResyncName(pods *corev1.PodList) (string, e
 	return "", nil
 }
 
-func isClusterEtcdOperatorReady(originalSpec *operatorv1.OperatorSpec) bool {
-	spec := originalSpec.DeepCopy()
-
-	// don't worry about errors.  If we can't decode, we'll simply stomp over the field.
-	existingConfig := map[string]interface{}{}
-	if err := json.NewDecoder(bytes.NewBuffer(spec.ObservedConfig.Raw)).Decode(&existingConfig); err != nil {
-		klog.V(4).Infof("decode of existing config failed with error: %v", err)
-	}
-
-	membersPath := []string{"cluster", "members"}
-	membersPendingPath := []string{"cluster", "pending"}
-
-	members, _, err := unstructured.NestedSlice(existingConfig, membersPath...)
+func (c *ClusterMemberController) isClusterEtcdOperatorReady() bool {
+	pendingMembers, err := c.EtcdList("pending")
 	if err != nil {
-		klog.Errorf("error reading members: %#v", err)
+		klog.Errorf("error getting pending members: %#v", err)
 		return false
 	}
-	klog.Infof("observed members: %#v", members)
-	pendingMembers, _, err := unstructured.NestedSlice(existingConfig, membersPendingPath...)
+	if len(pendingMembers) > 0 {
+		klog.Infof("some members are pending: %#v", pendingMembers)
+		return false
+	}
+	members, err := c.EtcdList("members")
 	if err != nil {
-		klog.Errorf("error reading pending members: %#v", err)
+		klog.Errorf("error getting members: %#v", err)
 		return false
 	}
-	if len(pendingMembers) > 0 {
-		klog.Infof("some members are pending: %#v", pendingMembers)
+	if len(members) == 0 {
+		klog.Infof("no etcd member found")
+		return false
+	}
+	if len(members) < 2 && c.IsMember("etcd-bootstrap") {
```

Comment:
Minor thing, but why `< 2`? Isn't `==1` is more readable? (we are already checking for `==0` above, so the only value we want to see is `1`, since length can never be negative).

Reply:
yes, it is, I will update it!

Reply:
updated